### DISCUSSION TITLE
SSL/TLS auth key-cert questions

### DISCUSSION THREAD

**Discussion by twizzter** [emqx/emqx#13362] [General]
Hi!
Taking my first steps in the topic of MQTT, and I have some doubts and questions in this regard.
I've been trying to connect to EMQX (serverless) using a SIM7000 modem. Running out of ideas. Support is not particularly helpful, modem itself generates only enigmatic "operation not allowed" message.
Seems that the problem is related to the secure connection (SSL-less connections work OK)
One of the possible clues might be: modem itself (unlike the larger SIM7600 model - where there are separately listed modes for connecting to the MQTT server: do not verify server/only verify server/verify both) has only one mode for connecting to the MQTT broker. I suspect that even if I want to do "verify server" i still need to have the client key and certificate generated so that the modem has a full set of files. 
Question is: if I use OpenSSL to generate a client key and sign it by myself, will these files "work" with the CA certificate downloaded from EMQX?

**Comment by id** [comment]
Hi @twizzter.

Please make sure you have added [Serverless CA](https://assets.emqx.com/data/emqxsl-ca.crt) into modem's trusted root certificates, and that you are using correct hostname assigned to your deployment.

**Reply by twizzter** [reply]
Thank you for the hint. 
- Hostname has been checked
- This CA has been downloaded and converted to modems internal format. 
- Modem's internal date/time is valid
- SNI added to modems configuration
The problem is SIM7000 datasheet says absolutely nothing how to setup a MQTT connection without client keys/certs - using CA cert only. 
Even MQTT example shows client cert and keys being used to estabilish a connection. That's why i asked about client keys/certs - maybe i need to generate them anyway. 
![obraz](https://github.com/emqx/emqx/assets/36953843/42c3514e-e4e4-4f56-9b6c-c3d721463183)

**Comment by zmstone** [comment]
I don’t think emqx serverless is configured to verify client certificates.
So you can go ahead generate a self-signed pair.

**Comment by zmstone** [comment]
> Question is: if I use OpenSSL to generate a client key and sign it by myself, will these files "work" with the CA certificate downloaded from EMQX?

If you mean to run emqx by yourself and use the default certificates came in the installation packages? then yes, server does not by default verify client certificate, and client is not required to send a certificate either.
